src/Parser/ShadowPriest/Modules/Spells/CallToTheVoid.js

```diff
@@ -0,0 +1,74 @@
+// import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Module from 'Parser/Core/Module';
+
+// import Icon from 'common/Icon';
+// import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+// import { formatThousands, formatNumber } from 'common/format';
+
+import makeWclUrl from 'common/makeWclUrl';
+
+
+class CallToTheVoid extends Module {
+  _damageDone = 0;
+  _generatedInsanity = 0;
+
+  _tentacles = {};
+
+  on_initialized() {
+    if (!this.owner.error) {
+      this.active = this.owner.selectedCombatant.traitsBySpellId[SPELLS.CALL_TO_THE_VOID_TRAIT.id];
+      this.load();
+    }
+  }
+
+  get damageDone(){
+    return this._damageDone;
+  }
+
+  get insanityGenerated(){
+    return this._generatedInsanity;
+  }
+
+  get insanityGeneratedPerSecond(){
+    return this.insanityGenerated / (this.owner.fightDuration / 1000);
+  }
+
+  load() {
```

Comment:
It's not ideal to have another API call that is always executed when a log is parsed. The WCL API rate limit if fairly limiting and it's important to keep calls down to avoid requiring API call queuing. 

Is there no other way of doing this, e.g. by checking the pets in the fights response and tracking those manually with the `damage` events? Player pet events should be in the API response.

Reply:
I will try and see if I can prevent the additional calls.